src/auth.service.js

```diff
@@ -192,14 +218,48 @@
       /* this triplication of the dependencies is error prone and i
        * don't see a reason for it. It would be nice to eliminate this
        * eventually - francesco 2016-10 */
-      return new UserManagementAuthService(
+      var service = new UserManagementAuthService(
         options,
         restangular,
         $log,
         $q,
         $rootScope,
-        $window
+        $window,
+        EHA_USER_MANAGEMENT_AUTH_UNAUTHORISED_EVENT,
+        EHA_USER_MANAGEMENT_AUTH_UNAUTHENTICATED_EVENT
       );
+
+      function authorisationPolicy (f) {
+        return function (policyArgument) {
+          service
+            .getCurrentUser()
+            .then(function (user) {
+              var authorised = f(user, policyArgument)
+              if (authorised) {
+                return p;
+              } else {
+                service.trigger(EHA_USER_MANAGEMENT_AUTH_UNAUTHORISED_EVENT);
+                return $q.reject(EHA_USER_MANAGEMENT_AUTH_UNAUTHORISED_EVENT);
+              }
+            })
+        }
+      }
+
+      /* used in the call centre - francesco 11-2016 */
+      service.requireUserWithAnyRole = authorisationPolicy(function (user, roles) {
+        return roles.forEach(function (authorised, role) {
+          return authorised || user.hasRole(role);
+        }, false);
+      })
+
+      /* used in the call centre - francesco 11-2016 */
+      service.anyRoleExcept = authorisationPolicy(function (user, exclude) {
+        return options.userRoles.forEach(function (authorised, role) {
+          return authorised || (role !== exclude && user.hasRole(role));
+        }, false);
+      })
```

Comment:
Do you have any test for these two new options? :thinking: 

Don't you want to say `reduce` instead of `forEach`